### PR TITLE
Update PPP

### PR DESCRIPTION
**Pull request by Criss-Wang**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -617,8 +617,11 @@ warehouse/supplier with the updated product. The method then updates the model w
    The method then passes a `CommandResult` with a success message back to the `LogicManager`. Finally, the model
 is saved and the GUI is updated with the success message.
 
-The following activity diagram summarizes what happens when a user updates a product:
-![Update Product Command Activity Diagram](images/UpdateCommandActivityDiagram.png)
+    The method then passes a `CommandResult` with a success message back to the `LogicManager`. Finally, the model
```
Comment (qlchan24) [inline docs/DeveloperGuide.md]: This is repeated from above

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -743,7 +758,7 @@ The list macros feature allows users to be able to view all presently saved macr
 
 #### How it is implemented
 
-The list macros feature is facilitated by the ListMacroCommand, whose job is to retrieve to list of macros and process it into a suitable format to be displayed to the user.
+The list macros feature is facilitated by the `ListMacroCommand`    , whose job is to retrieve to list of macros and process it into a suitable format to be displayed to the user.
```
Comment (qlchan24) [inline docs/DeveloperGuide.md:780]: The extra space seems unneccessary

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -687,8 +690,16 @@ If that is true, an exception will be thrown. This will be shown on the GUI as a
    The `AssignMacroCommand#execute` then passes a `CommandResult` with a success message back to the `LogicManager`. Finally, the model is saved and the GUI is updated with the success message.
 The user now updates the quantity of the product "Panadol" in the aforementioned warehouse by simply executing the command `uwm pd/Panadol`.
 
-The following activity diagram summarizes what happens when a user assigns a macro:
-![Assign Macro Command Activity Diagram](images/AssignMacroCommandActivityDiagram.png)
+    The `AssignMacroCommand#execute` method will then be called with the `model` instance. The method will first check if there is any existing macro in the model that uses the same alias.
```
Comment (qlchan24) [inline docs/DeveloperGuide.md]: This is also repeated above

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -732,8 +743,12 @@ retrieved macro will be removed from the model.
 
    The `RemoveMacroCommand#execute` then passes a `CommandResult` with a success message back to the `LogicManager`. Finally, the model is saved and the GUI is updated with the success message.
 
-The following activity diagram summarizes what happens when a user updates a product:
-![Remove Macro Command Activity Diagram](images/RemoveMacroCommandActivityDiagram.png)
+1. Result Display <br>
```
Comment (qlchan24) [inline docs/DeveloperGuide.md:765]: This is the same issue

**Review by qlchan24 — COMMENTED** [review]
Some issues as stated, otherwise LGTM

**Review by jeffreytjs — COMMENTED** [review]
Good pick up one the typos in DG. Do address Qin Liang's comments then it's good to go!

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -543,11 +544,11 @@ In the following section, the interaction between different objects will be disc
 
    After receiving an input from user for edit command, `EditCommandParser#parse` will be invoked.
 
-   As mentioned in above section, if either one of the compulsory prefixes are missing, `ParseException` will be thrown to remind users. Furthermore, invalid values supplied for `type` and `index` (e.g. `i/test`), a `ParseException` will be thrown.
+   As mentioned in above section, if either one of the compulsory prefixes is missing, `ParseException` will be thrown to remind users. Furthermore, invalid values supplied for `type` and `index` (e.g. `String` value for `index`), a `ParseException` will be thrown.
```
Comment (tohyuting) [inline docs/DeveloperGuide.md]: I think this should be i/test instead, since we mentioned that String may not be clear :)

**Review by tohyuting — COMMENTED** [review]
Other than the issues pointed out my Qin Liang, just one more comment. Otherwise, LGTM.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (Criss-Wang) [reply docs/DeveloperGuide.md]: Okay I think it's because I didn't check the content carefully, thanks!

**Review by qlchan24 — APPROVED** [review]
LGTM

**Pull request merged**